### ISSUE TITLE
Feature request: Make rope theta configurable

### ISSUE THREAD

**Issue by creatorrr** [axolotl-ai-cloud/axolotl#810]
### ⚠️ Please check that this feature request hasn't been suggested before.

- [X] I searched previous [Ideas in Discussions](https://github.com/OpenAccess-AI-Collective/axolotl/discussions/categories/ideas) didn't find any similar feature requests.
- [X] I searched previous [Issues](https://github.com/OpenAccess-AI-Collective/axolotl/labels/enhancement) didn't find any similar feature requests.

### 🔖 Feature description

Make `rope_theta` configurable in the config with a default value of `10_000.0` (which is the default for both llama and mistral).

### ✔️ Solution

This paper [Scaling Laws of RoPE-based Extrapolation](https://arxiv.org/abs/2310.05209) argues that:

> We first highlight a surprisingly strange phenomenon 10000 is the worst base value for
> RoPE-based extrapolation in the fine-tuning phase. Remarkably, fine-tuning with either
> a larger or smaller base within the training context length greatly enhances extrapolation,
> which provides a new vision to the extrapolation research of RoPE.

I'd love to try other values suggested in the paper but right now it's not possible to configure with axolotl.

### Acknowledgements

- [X] My issue title is concise, descriptive, and in title casing.
- [X] I have searched the existing issues to make sure this feature has not been requested yet.
- [X] I have provided enough information for the maintainers to understand and evaluate this request.

**Comment by NanoCode012** [comment]
Would you be interested in making this PR?

**Comment by creatorrr** [comment]
I would love to but it'd probably take me some time since I am really caught up right now. So if anyone else wants to pick it up, please go ahead. Otherwise, I'll def come around to it in a week or two. :)

**Comment by adi-kmt** [comment]
@NanoCode012 could take this up, if you want

**Comment by NanoCode012** [comment]
Hey @kamathis4 , thanks for the offer. I didn't revisit this Issue in a while.

It is now possible to override any model config as shown below. Feel free to let me know if any other Issue interests you!

```yaml
# optional overrides to the base model configuration
model_config:
  rope_theta: # some value
```

If I missed anything, feel free to let me know.
